tests/xoreutils/test_which.py

```diff
@@ -1,14 +1,15 @@
 import os
 import tempfile
+from unittest import TestCase
 
 from xonsh.tools import ON_WINDOWS
 from xonsh.xoreutils import _which
 
 
-class TestWhich:
+class TestWhich(TestCase):
     # Tests for the _whichgen function which is the only thing we
     # use from the _which.py module.
-    def setup(self):
+    def setUp(self):
```

Comment:
```suggestion
    @classmethod
    def setup_class(cls):
```

I think this is the standard way to handle class-wrapped setup and teardown in pytest (will require swapping out all the `self` below, though).  Not a blocker -- I am not a fan of test classes anyway.

Reply:
setup_class will run at class level only during the start of the test. It is a not straight equivalent to the setup of nosetest . For now we can use the unit-test and later refactor to a pytest if needed.